### ISSUE TITLE
Set ignore path in .eslintrc

### ISSUE THREAD

**Issue by remcohaszing** [eslint/eslint#3529]
The ESLint docs show an example of how to use `.gitignore` as the ignore path.

This seems a very useful feature to me. However, my editors don't support this.

It would be useful to specify the ignore-path in `.eslintrc` instead of as a command line parameter. This way editors or whatever tools using ESLint don't have to supply this.

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by lo1tuma** [comment]
I don’t think this is possible, because you can have multiple `.eslintrc` files but only one ignore file.

**Comment by ilyavolodin** [comment]
Also .eslintrc's are constructed during traversing, so being able to specify it in .eslintrc would not be possible at all. You can, however, specify ignore pattern (or ignore path) on CLI.

**Comment by nzakas** [comment]
We investigated this before and determined it would be a lot of work for something that we already have via `.eslintignore`. I'd suggestion contacting the editor/plugin author to ask them to add support for `.eslintignore`.

**Comment by jaydenseric** [comment]
You can [set `ignorePath` within stylelint config](http://stylelint.io/user-guide/configuration/#stylelintignore). They went to the trouble to switch to node-ignore to [facilitate pointing `ignorePath` to .gitignore](https://github.com/stylelint/stylelint/issues/1452). 

As time goes on we find ourselves linting all the things. Having separate .gitignore, .eslintignore, .stylelintignore, etc. files that contain the same rules is tedious. Junior devs generally know to update .gitignore when messing with the project but always forget the separate linting configs.

It would be awesome if eslint supported setting `ignorePath` in config and not just via CLI, then 3 files would become one. The growing number of config files floating around our project roots is tiresome, internally we have a policy of consolidating all config possible in package.json.

The only thing preventing the dream of all linting config in package.json, all referencing .gitignore is this issue 😢

**Comment by moos** [comment]
Would be very nice to have!

**Comment by wouter2203** [comment]
This has my upvote!

**Comment by mherodev** [comment]
Wish this wasn't closed. We've managed to wrangle all of our dotfiles into their own directory except for this little guy.

**Comment by justin808** [comment]
You got my upvote!

**Comment by damianobarbati** [comment]
any news on this? How can I tell eslint to ignore the same files specified in my .gitignore in the package.json eslintConfig entry?

**Comment by onury** [comment]
@nzakas This should be re-opened. Multiple config files is not right.

BTW, referencing .gitignore does not solve it. You might have bundled dependencies to be commited to Git that are not to be linted.

**Comment by deryagin** [comment]
May be plain symlinks could help to solve this issue?
```bash
mv .gitignore .ignorepaths
ln -s .ignorepaths .gitignore
ln -s .gitignore .eslintignore

ls -A1
.eslintignore -> .ignorepaths
.gitignore -> .ignorepaths
.ignorepaths
.eslintrc.js
```

**Comment by timhuff** [comment]
I came here looking for a ignore property for `.eslintrc.js`. It hadn't occurred to me that I should make it my `.gitignore` file. If anyone wants to do this, I believe running `ln -s .gitignore .eslintignore` in project root should do the trick.
